go-controller/pkg/ovn/pods.go

```diff
@@ -236,31 +235,40 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) error {
 
 	var podMac net.HardwareAddr
 	var podIfAddrs []*net.IPNet
-	var args []string
-	var addresses string
+	var cmds []util.OVNCommandInterface
+	var addresses []string
+
+	// Check if the pod's logical switch port already exists. If it
+	// does don't re-add the port to OVN as this will change its
+	// UUID and and the port cache, address sets, and port groups
+	// will still have the old UUID.
+	lsp, err := oc.ovnNBClient.LSPGet(portName)
+	if err != nil && err != goovn.ErrorNotFound && err != goovn.ErrorSchema {
```

Comment:
I am not even sure why will we ever get `goovn.ErrorSchema`?

Reply:
this is to handle the case where LSPGet is called without any LSP existing in the cache. The go-ovn library maintains a local cache of rows for fast fetching and if there has been no row added to the `LogicalSwitchPort` table ever, that table is not in the cache, at which point, the library throws a `goovn.ErrorSchema`.